Vector2D.java

```diff
@@ -0,0 +1,66 @@
+package org.firstinspires.ftc.teamcode;
+
+/**
+ * Created by NoahR on 9/16/17.
+ */
+
+public class Vector2D {
+    private double m_nXComponent;
+    private double m_nYComponent;
+
+
+    public Vector2D( double iXComponent, double iYComponent ) {}
+
+    public double GetXComponent() { return m_nXComponent; }
+    public double GetYComponent() { return m_nYComponent; }
+
+    public double GetMagnitude()
+    {
+        return Math.sqrt( ( Math.pow(m_nXComponent, 2 ) ) + ( Math.pow(m_nYComponent, 2 ) ) );
+    }
+    public double GetDirection()
+    {
+        return Math.atan( m_nYComponent / m_nXComponent);
+    }
+
+    public void Set( Vector2D vNewVector ) { this.SetComponents( vNewVector.GetXComponent(), vNewVector.GetYComponent() ); }
+    public void Set( double nMagnitude, double nDirection )
```

Comment:
Maybe change name to `setPolar` or something to indicate that it takes polar coordinates.